services/cd-service/pkg/repository/transformer.go

```diff
@@ -1343,36 +1343,66 @@ func (u *DeleteEnvFromApp) Transform(
 		return "", err
 	}
 
-	fs := state.Filesystem
-	thisSprintf := func(format string, a ...any) string {
-		return fmt.Sprintf("DeleteEnvFromApp app '%s' on env '%s': %s", u.Application, u.Environment, fmt.Sprintf(format, a...))
-	}
+	if state.DBHandler.ShouldUseOtherTables() {
+		releases, err := state.DBHandler.DBSelectReleasesByApp(ctx, transaction, u.Application)
+		if err != nil {
```

Comment:
How do we filter by deleted releases? We shouldn't remove envs from deleted releases

Reply:
We keep track of what releases are active by application on an all_releases table

Reply:
Fixed